asr/base.py

```diff
@@ -97,63 +98,66 @@ def review(self):
             estimator=self.model,
             X_training=self.X[init_ind],
             y_training=init_labels,
+            query_strategy=usample_cached,
 
             # additional arguments to pass to fit
             **self.fit_kwargs)
 
         # remove the initial sample from the pool
         pool_ind = np.delete(pool_ind, init_ind)
 
+        self._logger.add_training_log(init_ind, init_labels, i=0)
         query_i = 0
 
         while not self._stop_iter(query_i, pool_ind):
-
+            pred_proba = []
             # Make a query from the pool.
             query_ind, query_instance = self.learner.query(
                 self.X[pool_ind],
-                n_instances=min(self.n_instances, len(pool_ind))  # ,
-                # verbose=self.verbose
+                n_instances=min(self.n_instances, len(pool_ind)),
+                pred_proba=pred_proba,
             )
 
+            # Log the probabilities of samples in the pool being included.
+            if len(pred_proba) > 0:
+                self._logger.add_pool_pred_prob(
+                    pool_ind, pred_proba[0], i=query_i
+                )
+
             # classify records (can be the user or an oracle)
             y = self._classify(query_ind)
-
             # train model
             self._prior_teach()
-
             # Teach the learner the new labelled data.
             self.learner.teach(
                 X=query_instance,
                 y=y,
-                only_new=False,  # check docs!!!!
+                only_new=False,  # Only query papers not already labeled.
 
                 # additional arguments to pass to fit
                 **self.fit_kwargs)
 
             # remove queried instance from pool
             pool_ind = np.delete(pool_ind, query_ind, axis=0)
 
-            # predict the label of the unlabeled entries in the pool
-            if len(pool_ind) > 0:
-                pred = self.learner.predict(self.X[pool_ind])
-
-            # add results to logger
-            self._logger.add_training_log(query_ind, y, i=query_i)
-            # self._logger.add_pool_log(pool_ind, pred, i=query_i)
-
-            # # reset the memory of the model
-            # self.learner._model.set_weights(init_weights)
+            # Add the query indexes to the log.
+            self._logger.add_training_log(query_ind, y, i=query_i+1)
```

Comment:
Shall we start `query_i` at 1?

Reply:
No, I think it's fine like this. In the same loop query_i is used (which would have to be replaced by query_i-1 in that case).